ag/task_test.go

```diff
@@ -0,0 +1,68 @@
+package ag_test
+
+import (
+	"testing"
+
+	pb "github.com/autograde/quickfeed/ag"
+	"github.com/google/go-cmp/cmp"
+	"google.golang.org/protobuf/testing/protocmp"
+)
+
+// TestHasChanged tests if HasChanged returns the correct value
+func TestHasChanged(t *testing.T) {
+	originalTask := &pb.Task{
+		ID:              1,
+		AssignmentID:    1,
+		AssignmentOrder: 1,
+		Title:           "This is the original task",
+		Body:            "Description description",
+		Name:            "lab1/1",
+	}
+
+	// Comparing the original task with itself.
+	wantResult := false
+	gotResult := originalTask.HasChanged(originalTask)
+
+	if diff := cmp.Diff(wantResult, gotResult, protocmp.Transform()); diff != "" {
```

Comment:
It is a bit overkill to use `cmp.Diff` on bool results since it is easy to compare with plain Go:

```go
if want != got {
  t.Errorf("%v.HasChanged(%v) = %t, expected %t", originalTask, originalTask, got, want)
}
```
Note that the two inputs may print too much, so maybe that's where you would want to compare the two tasks instead of the bool values... Not sure what is best here... I guess what allows to see why it might fail.

This type of test might lend itself to so called [table-driven tests](https://github.com/golang/go/wiki/TableDrivenTests) because you have multiple different task examples to compare; which could be defined separately. Then use something like this in the Go test struct:

```go
inTask1, inTask2 pb.Task
want bool
```

Reply:
I thinks perhaps you can simplify the input tasks to the bare minimum to test the method in question, namely only define tasks with various configurations of the `Title` and `Body` fields, and ignore all the other fields... Thus the output using the `%v` verb would not be too verbose. And you can set the title and body fields to be short strings to avoid verbosity in the output.